lib/agent.js

```diff
@@ -233,6 +237,12 @@ Agent.prototype.captureError = function (err, opts, cb) {
     prepareError(parsers.parseMessage(err))
   } else {
     parsers.parseError(err, agent, function (_, error) {
+      const hasInternal = error.exception.stacktrace
+        .filter(isInternalFrame)
```

Comment:
You can optimize this by using `some` instead of `filter`. That way it will stop when it finds the first occurrence of an internal frame, e.g:

```js
const hasInternal = error.exception.stacktrace.some(internalFrame)
```

Reply:
Good call.

Reply:
At second thought, do we really want to look for this frame _any_ where in the stack? We do a lot of monkey patching and we're probably going to be present in a bunch of stacks even though the error didn't originate from within the agent. Maybe we should just look for the stop stack 🤔 Hmmm this quickly gets tricky

Reply:
What do we do if the failure originates from one of our dependencies then? 🤔

Reply:
Switched to `some(...)`. It's an _improvement_ of the current logic, but seems like it should maybe be different somehow...